service/storage_service.cc

```diff
@@ -5523,6 +5523,8 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(raft
             state.last_index = cmd_index;
         }
 
+        co_await _group0->client().wait_until_group0_upgraded(_group0_as);
```

Comment:
I think it makes sense to do it even earlier -- before we do **anything** on group 0, even the `read_barrier` which is done above -- so why not do it at the very beginning of this function

Reply:
done, moved to the beginning of the `raft_topology_cmd_handler`